### ISSUE TITLE
jlink-image fails on Helidon quickstart MP example with Java 17 when creating CDS archive

### ISSUE THREAD

**Issue by barchetta** [oracle/helidon-build-tools#537]
*  helidon-maven-plugin 2.3.0 (Helidon 2.4.0)
* Apache Maven 3.8.2
* Java: java 17.0.1 2021-10-19 LTS
* OS: MacOS and Linux

Generating the jlink image fails when using Java 17 with the `helidon-quickstart-mp` example (it works with the SE quickstart example):
```
[INFO] Installing 146 application jars in target/helidon-quickstart-mp-jri/app
[INFO]   checking index in CDI beans archive helidon-quickstart-mp.jar
[INFO]   creating missing index for CDI beans archive helidon-microprofile-server-2.4.0-SNAPSHOT.jar
[INFO]   creating missing index for CDI beans archive helidon-health-checks-2.4.0-SNAPSHOT.jar
[INFO]   creating missing index for CDI beans archive helidon-microprofile-fault-tolerance-2.4.0-SNAPSHOT.jar
[INFO]   creating missing index for CDI beans archive jersey-weld2-se-2.35.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:08 min
[INFO] Finished at: 2021-10-29T08:45:08-07:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.helidon.build-tools:helidon-maven-plugin:2.3.0:jlink-image (jlink-image) on project helidon-quickstart-mp: Image generation error: io.helidon.build.util.ProcessMonitor$ProcessFailedException: Creating Class Data Sharing archive for helidon-quickstart-mp.jar failed with exit code 134 -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```
Looks like this is caused by a SEGV in jvm when creating the Class Data Sharing archive:

```
. . .
[DEBUG] Executing command: /Users/jdipol/GitHub/barchetta/helidon/examples/quickstarts/helidon-quickstart-mp/target/helidon-quickstart-mp-jri/bin/java -Dexit.on.started=! -Xshare:dump -XX:SharedArchiveFile=lib/start.jsa -XX:SharedClassListFile=/var/folders/xs/54nvktvd6f9dn9xrvtt1bpw80000gn/T/start193575815125357631.classlist -Dfile.encoding=UTF-8 -jar app/helidon-quickstart-mp.jar
[DEBUG] Process ID: 16590
[DEBUG] Waiting for completion, pid=16590, timeout=60, unit=SECONDS
. . .
[5.143s][warning][cds] Skipping io/netty/handler/ssl/ReferenceCountedOpenSslServerContext: Failed verification
[5.143s][warning][cds] Skipping io/netty/handler/ssl/OpenSslServerContext: Failed verification
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x000000010ddb6e47, pid=16590, tid=17411
#
# JRE version: Java(TM) SE Runtime Environment (17.0.1+12) (build 17.0.1+12-LTS-39)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (17.0.1+12-LTS-39, interpreted mode, tiered, compressed oops, compressed class ptrs, g1 gc, bsd-amd64)
# Problematic frame:
# V  [libjvm.dylib+0x9b6e47]  _ZN22SystemDictionaryShared36adjust_lambda_proxy_class_dictionaryEv+0x47
#
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/jdipol/GitHub/barchetta/helidon/examples/quickstarts/helidon-quickstart-mp/target/helidon-quickstart-mp-jri/hs_err_pid16590.log
#
# If you would like to submit a bug report, please visit:
#   https://bugreport.java.com/bugreport/crash.jsp
#

```

**Comment by batsatt** [comment]
I was able to reproduce this failure and have filed a bug against the JVM; it was assigned an "internal review ID" of 9071851 which should get converted to a bug soon. I also pinged the CDS developer directly and let him know about this bug.

**Comment by batsatt** [comment]
This issue may already have been fixed: https://bugs.openjdk.java.net/browse/JDK-8274944

**Comment by batsatt** [comment]
I have verified that the fix for https://bugs.openjdk.java.net/browse/JDK-8274944 does fix this issue; and should be in JDK 17.0.3.

**Comment by barchetta** [comment]
I have confirmed this is also fixed in an early build of 18 (build 18-ea+21-1341)

**Comment by barchetta** [comment]
And I have confirmed this is fixed in 17.0.3 2022-04-19 LTS
